lib/IDE/SyntaxModel.cpp

```diff
@@ -706,7 +708,11 @@ std::pair<bool, Expr *> ModelASTWalker::walkToExprPre(Expr *E) {
       llvm::SaveAndRestore<ASTWalker::ParentTy> SetParent(Parent, E);
       subExpr->walk(*this);
     }
-    return { false, walkToExprPost(SE) };
+    auto result = walkToExprPost(SE).Value;
+    if (!result)
+      return Action::Stop();
```

Comment:
We could add `StopIfNullptr` though a couple of the callback cases do `SkipChildren` instead of `Continue`, so we'd also need something for that, and I'm not sure it would pull its weight (ideally we'd propagate explicit walker actions into the clients of those walkers instead of relying on `nullptr` returns).